setup.py

```diff
@@ -33,6 +33,7 @@
         'ecmwf-api-client',
         'eofs',
         'esmvalcore>=2.3.1,<2.4',
+        'esmpy',
```

Comment:
Not to be a PITA, but now this changes the capitalization. I am not sure that will work.

Reply:
yes, in fact we should de-capitalize all the deps, pip uses lower case anyways, but it doesn't mind if they're uppers

Reply:
Interesting. Do you have a reference for that? I'd like to read more, particularly since in this case the package is not distributed via PyPI. I always thought the capitalization was just a convention followed by (almost) all packages in the `name` argument to `setup`.

Reply:
that's a very good question - I was under the assumption that the case is not important since they are to be used part of links to grab and download the packages. Here is a good thread, it looks like pip 20.2 broke the case insensitive nature but then they fixed it https://gitlab.com/gitlab-org/gitlab/-/issues/244412 (also check out [PEP-426](https://www.python.org/dev/peps/pep-0426/#name) )

Reply:
Well, the first link is just about something gitlab does. If you have a gitlab server, it comes with its own, built-in pypi server that allows you to make packages distributable in your organization without publishing them to the big PyPI.

The PEP has status "withdrawn".

So I am afraid both are not applicable here.

Reply:
totally missed the Status: withdrawn, good spot! This one is active, although it is for in-code naming but I am assuming the package name convention should still apply https://www.python.org/dev/peps/pep-0008/#package-and-module-names - should ask @bouweandela - he's named `esmvalcore` this way and not `ESMValCore` for a reason :+1:

Reply:
Sorry, that was just my personal preference, nothing official.

Reply:
@valeriupredoi, that link seems to talk only about modules and packages, and for them only about a convention. This is about projects, which often do not follow that convention, with both various capitalizations and unusable characters like `-` being common.

Certainly upstream does not follow that convention and I have not seen anything that indicates one should use a different spelling than upstream.

Reply:
OK man, I reinstated the capitalization, I need money to hire a PhD student do some research on the capitalization of PyPi package names :laughing: